### PR TITLE
Allow setting controller image repository and tag independently

### PR DESCRIPTION
**Pull request by j2L4e**
In order to incorporate this extension into [yake](https://github.com/YAKEcloud/yake) an attempt was made to add its chart to the [gardener-charts](https://github.com/gardener-community/gardener-charts) helm repo. The chart importer as well as yake itself assume that image repositories and versions can be set independently.

This once worked but was removed [when adding the skaffold setup](https://github.com/stackitcloud/gardener-extension-shoot-flux/pull/12/files).

This PR restores the old behavior while keeping the possibility to use a plain string image source to be skaffold compatible.

cc @maboehm

### CODE REVIEW

File: charts/gardener-extension-shoot-flux/templates/_helpers.tpl [hunk 1]
```diff
@@ -13,3 +13,13 @@ app.kubernetes.io/name
 {{ include "labels.app.key" . }}: {{ include "labels.app.value" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end -}}
+
+{{- define "image" -}}
+  {{- if eq (printf "%T" .Values.image) "string" }}
```
Comment (maboehm) [inline charts/gardener-extension-shoot-flux/templates/_helpers.tpl:18]: I'd prefer
```suggestion
  {{- if kindIs "string" .Values.image }}
```
https://helm.sh/docs/chart_template_guide/function_list/#kind-functions

Any particular reason you used `printf "%T"`?

Comment (j2L4e) [reply charts/gardener-extension-shoot-flux/templates/_helpers.tpl]: Nope, I was unaware that helm brings `kindIs`. Neat.

**Review by maboehm — APPROVED** [review]
thanks!

**Comment by j2L4e** [comment]
Great, thanks! Do you mind making a release?

**Pull request merged**